### PR TITLE
Use Sass to Loop and Generate Class-based Rules

### PR DESCRIPTION
**Pull request by talbs**
This work converts some of the class-based rules in the UXPL to:

* be set using simple configuration within each pattern's partial
* be generated using Sass's ``@each`` method

The goal is to reduce redundant code and to help manage sizes/parameters in a single place.

- - -

##### Reviewers

* [ ] FED - @frrrances

### CODE REVIEW

File: public/css/main-rtl.css [hunk 1]
```diff
@@ -5208,27 +5208,27 @@ Note: sure .FONT_NAME comes first for adjoining classes bug in IE7.
   }
 }
 
-.depth--3, .pldoc-wrapper-content-supplemental {
+.btn--3, .pldoc-wrapper-content-supplemental {
   background: #282c2e;
 }
 
-.depth--2 {
+.btn--2 {
```
Comment (frrrances) [inline public/css/main-rtl.css]: did you mean for this to change from depth to btn (here and below)?

File: public/css/main-ltr.css [hunk 2]
```diff
@@ -5208,27 +5208,27 @@ Note: sure .FONT_NAME comes first for adjoining classes bug in IE7.
   }
 }
 
-.depth--3, .pldoc-wrapper-content-supplemental {
+.btn--3, .pldoc-wrapper-content-supplemental {
```
Comment (frrrances) [inline public/css/main-ltr.css]: did you mean for this to change from depth to btn (here and below)?

File: pattern-library/sass/patterns/_depth.scss [hunk 3]
```diff
@@ -83,30 +91,10 @@ $depth-2-bg:    palette(grayscale, white)   !default;  // +2 depth level
...
-}
-
-.depth-1 {
-    @extend %depth-1;
-}
-
-.depth-2 {
-    @extend %depth-2;
+@each $depth-level in $depth-levels {
+    .btn-#{$depth-level} {
```
Comment (frrrances) [inline pattern-library/sass/patterns/_depth.scss]: I think this is the problem, now that I look at it... .btn -> .depth

**Comment by frrrances** [comment]
@talbs This makes it easy to maintain lots of canned classes but we loose a lot of readability, especially around the canned classes that makes it hard to use. I don't expect we'll be adding a ton more canned classes (like we have to do with images), so I'm not sure how much gain we're getting here. Thoughts?

File: pattern-library/sass/patterns/_depth.scss (hunk 3 above)
Comment (talbs) [inline pattern-library/sass/patterns/_depth.scss]: Yep, thanks for the good catch. Copying and pasting got away from me.

**Comment by talbs** [comment]
Gonna close this for now as I think we have larger things to worry about in and out of the UXPL. We can revisit if canned application styling becomes a pain.

**Pull request closed without merging**